src/depackers/vorbis.c

```diff
@@ -3341,9 +3335,6 @@ static int vorbis_decode_packet_rest(vorb *f, int *len, Mode *m, int left_start,
       int r;
       uint8 do_not_decode[256];
       int ch = 0;
-
-      memset(do_not_decode, 0, 256);
-
```

Comment:
1) `do_not_decode` and `ch` are declared in `vorbis_decode_initial` (line 3342). `ch` has an initial value of 0.
2) A loop on `f->channels` determines which channel to decode according to a map. Before `ch` is incremented, `do_no_decode[ch]` is initialized.
3) `decode_residue` is called with `ch` and `do_not_decode` being passed to parameters with identical names (line 3360).
4) Usages of `do_not_decode` in `decode_residue`:
    1) Line 2161: iterated with `i` which is bounded by `ch`.
    2) Line 2166: iterated with `j` which is bounded by `ch`.
    3) Line 2270: iterated with `j` which is bounded by `ch`.
    4) Line 2288: iterated with `j` which is bounded by `ch`.
5) Indices of `do_not_decode` >= `ch` do not need to be initialized and indices <`ch` already are being initialized.

Introduced by this patch: https://github.com/libxmp/libxmp/commit/3b6214b3b9e138eba14d0725bd8e99853c7b1786.